app/coffee/ResourceStateManager.coffee

```diff
@@ -11,18 +11,19 @@ module.exports = ResourceStateManager =
 	# incremental update to be allowed.
 	#
 	# The initial value is passed in and stored on a full
-	# compile.
+	# compile, along with the list of resources..
 	#
 	# Subsequent incremental compiles must come with the same value - if
-	# not they will be rejected with a 409 Conflict response.
+	# not they will be rejected with a 409 Conflict response. The
+	# previous list of resources is returned.
 	#
 	# An incremental compile can only update existing files with new
 	# content.  The sync state identifier must change if any docs or
 	# files are moved, added, deleted or renamed.
 
 	SYNC_STATE_FILE: ".project-sync-state"
 
-	saveProjectStateHash: (state, basePath, callback) ->
+	saveProjectStateHash: (state, resources, basePath, callback = (error) ->) ->
```

Comment:
This should now be renamed to `saveProjectState` or similar, since it does more than just the hash now.